src/crawlers/browser_crawler.js

```diff
@@ -102,7 +102,8 @@ class BrowserCrawler extends BasicCrawler {
      * @ignore
      */
     async _handleRequestFunction(crawlingContext) {
-        const page = await this.browserPool.newPage();
+        const { id } = crawlingContext;
+        const page = await this.browserPool.newPage({ contextId: id });
```

Comment:
It should be just `id`, not `contextId`. Basically, you're giving the page a custom ID.

Reply:
Ok, thinking about it, given the comments below, I'm not sure now. Basically there are two options.

1. Allow users to set anything to `launchContext` via `newPage`, but that's a bit weird, because there could be a hundred `newPage` calls for a single `launchContext` (single browser). And where would all the things go?

2. Allow only specific things like `id`. Then, internally in `browser-pool` handle the problem with multiple pages per context. Such as keeping a dynamic list of page ids on the context.

For our use case, both would work. It does not really matter whether we set and retrieve `contextId` or `id`.